### ISSUE TITLE
2.10-dev-20140827 crash at startup

### ISSUE THREAD

**Issue by jens-maus** [jens-maus/yam#582]
**Originally by _AmigaSurfer@t-online.de_ on 2014-09-04 22:32:00 +0200**

___
## Steps to reproduce
 1. start YAM 2.10-dev-20140827
 2. wait

## Expected results
Requester with message about YAM 2.10 is beta.

## Actual results
WinUAE crash

## Regression
YAM-dev-20140809 does not crash.

## Notes
YAM.debug crash with WInUAEenforcer, too. Whole WinUAE is frozen.
The last line in sashimi.log is:
00:D|00:00:00.000000|  YAM.c:234:InitLib: library xpkmaster.library v5.2 successfully opened.
The last line in winuae.log is:
17-568 [1482 225-072]: B-Trap f058 at 10279e0a (98129E0A)

YAM.debug did not crash with MuForce and MuGuardianAngel. The beta-Requster appears. But then the message, that YAM is out of memory. I have already increased the memory from 512 MByte to 1024 MByte.
The sashimi.log shows some errors.

Both logs are attached.

**Comment by jens-maus** [comment]
**Originally by _AmigaSurfer@t-online.de_ on 2014-09-04 22:32:34 +0200**

___
**Attachment added:** [`winuaelog YAM.debug 2.10dev2014-09-27 Enforcer.zip`](https://yam.ch/attachments/582/winuaelog%20YAM.debug%202.10dev2014-09-27%20Enforcer.zip) (7.8 KiB)
Enforcer-Log

**Comment by jens-maus** [comment]
**Originally by _AmigaSurfer@t-online.de_ on 2014-09-04 22:33:16 +0200**

___
**Attachment added:** [`winuaelog YAM.debug 2.10dev2014-09-27 MuForce MuGuardian.zip`](https://yam.ch/attachments/582/winuaelog%20YAM.debug%202.10dev2014-09-27%20MuForce%20MuGuardian.zip) (10.4 KiB)
MuForce MuGuardianAngel Log

**Comment by tboeckel** [comment]
**Originally on 2014-09-05 08:35:01 +0200**

___
I cannot reproduce this issue. For me that nightly build version starts up without any problem on my WinUAE setup.

Did you try the previous nightly builds between 2014-08-09 and 2014-08-27? They can be downloaded from http://nightly.yam.ch

**Comment by jens-maus** [comment]
**Originally by _AmigaSurfer@t-online.de_ on 2014-09-08 23:25:28 +0200**

___
I tried YAM.debug from version 2014-08-27 to 2014-08-09 and then 2014-09-06 without WBStartup, but with sashimi, wipeout and winuaeenforcer. The crashes now appears after I accept the beta-requester. 2014-08-27 hangs whole WinUAE while the progress bar shows "Warte auf Benutzeranmeldung" but the buttons for the two accounts are not visible yet. All other versions did crash WinUAE while the progress bar shows "Initialisiere Bibliotheken und Klassen". I had to kill WinUAE with the task manager.
2014-08-09 did not crash.
Latest 2014-09-06 did crash.
In all cases I quit WinUAE after copying the corresponding YAM.debug and start WinUAE again, to get a clean system with no other program running before.

YAM.debug 2.10dev-2014-08-27 B-Trap fa72
YAM.debug 2.10dev-2014-08-26 B-Trap f9c2
YAM.debug 2.10dev-2014-08-20 B-Trap fc2a
YAM.debug 2.10dev-2014-08-19 B-Trap f93a
YAM.debug 2.10dev-2014-08-18 B-Trap fe0a
YAM.debug 2.10dev-2014-08-17 Illegal instruction 0fbe
YAM.debug 2.10dev-2014-08-16 Illegal instruction c87e
YAM.debug 2.10dev-2014-08-09 no Crash

YAM.debug 2.10dev-2014-09-06 Exception 3

Then I tried the latest version 2014-09-06 with MuFurce and MuGuardianAngel. I used WinUAE64 for this, to get 2 GByte of FastRAM.
After a while there were only 410 MByte FastRAM left. I waited about 15 minutes, but the amount of free RAM didn't change. Except if I close a window at the workbench, then a little bit RAM was freed. Then I quit WinUAE. Maybe YAM.debug was in an endless loop? The sashimi.log is really big (15 MByte)!

**Comment by jens-maus** [comment]
**Originally by _AmigaSurfer@t-online.de_ on 2014-09-08 23:28:40 +0200**

___
**Attachment added:** [`winuaelog YAM.debug 2.10dev-2014-09-08 ohne WBStartup Enforcer Exception 3.zip`](https://yam.ch/attachments/582/winuaelog%20YAM.debug%202.10dev-2014-09-08%20ohne%20WBStartup%20Enforcer%20Exception%203.zip) (7.8 KiB)

**Comment by jens-maus** [comment]
**Originally by _AmigaSurfer@t-online.de_ on 2014-09-08 23:29:55 +0200**

___
**Attachment added:** [`winuaelog YAM.debug 2.10dev-2014-09-08 ohne WBStartup MuForce.zip`](https://yam.ch/attachments/582/winuaelog%20YAM.debug%202.10dev-2014-09-08%20ohne%20WBStartup%20MuForce.zip) (308.4 KiB)

**Comment by jens-maus** [comment]
**Originally by _AmigaSurfer@t-online.de_ on 2014-09-08 23:33:36 +0200**

___
I added both logs (enforcer and MuForce) with latest nightly build 2014-09-06.
There is an error in the file names. It is 2014-09-06, not 2014-09-08.

**Comment by jens-maus** [comment]
**Originally by _AmigaSurfer@t-online.de_ on 2014-09-09 00:01:25 +0200**

___
**Attachment added:** [`winuaelog YAM.debug 2.10dev-2014-09-06 ohne WBStartup MuFurce ohne Wipeout.zip`](https://yam.ch/attachments/582/winuaelog%20YAM.debug%202.10dev-2014-09-06%20ohne%20WBStartup%20MuFurce%20ohne%20Wipeout.zip) (7.7 KiB)

**Comment by jens-maus** [comment]
**Originally by _AmigaSurfer@t-online.de_ on 2014-09-09 00:06:04 +0200**

___
I remembered now, that you told me some time ago to not use wipeout together with MuForce and MuGuardian. I made a new log. Version 2014-09-06 seems to freeze the mouse pointer now while the progress bar shows "Warte auf Benutzeranmeldung...". I waited some time. Then I opened the WinUAE config window and quit WinUAE64 with F11. This was still working.
The new log is attached. There seems to be no error in the log. :(

**Comment by tboeckel** [comment]
**Originally on 2014-09-09 06:21:18 +0200**

___
Since the nightly build of 2014-08-09 is the last working one I suspect my change done on 2014-08-15 to be the one causing the trouble. It is just a simple one which allows to enter the user password immediately instead of having to activate the string object first manually.

Are you still using MUI 3.8 or MUI 3.9 already? If the former is the case, then this is the proof that MUI 3.8 does not yet support asynchronous methods delayed by a certain amount of time. If you are using MUI 3.9 already then we will have to investigate further.

You should have mentioned that bit of information about the multiple users a bit earlier. Your steps for reproduction are far too scarce in this case. Usually people have just a single user configured in YAM. The information about multiple users is really important here. Please provide such information from the beginning next time.

**Comment by tboeckel** [comment]
**Originally on 2014-09-09 06:26:11 +0200**

___
Regarding the logs all I can say that there is something really fishy going on on your system again. See this line:


```
00:D|00:00:00.000000|  NewReadArgs.c:441:CS_Buffer[1669537988]: '
'
```


The number in the squared brackets is the length of either the command line or all ToolTypes of YAM's icon. Usually this is a quite small number, most of the time far less than 1000. In your case it encodes a line consisting of 1.6GB!! This is something that cannot be true.

**Comment by tboeckel** [comment]
**Originally on 2014-09-09 07:47:58 +0200**

___
In (70483b9):
* mui/YAMApplication.c: added a workaround for MUI 3.8 which does not yet support delayed execution of pushed methods. The delay time must be stripped from the parameter counter. Otherwise it will be treated as an arbitrarily large number of parameters which might cause all kinds of weird behaviour to happen. This closes #582.

**Comment by jens-maus** [comment]
**Originally by _AmigaSurfer@t-online.de_ on 2014-09-10 20:35:39 +0200**

___
Thank you. The new nightly build does not crash at startup.
And yes, I'm still using MUI 3.8.
